src/FrameDecorator/AbstractFrameDecorator.php

```diff
@@ -749,11 +766,9 @@ public function split(Frame $child = null, bool $force_pagebreak = false)
 
         $this->get_parent()->split($split, $force_pagebreak);
 
-        // If this node resets a counter save the current value to use when rendering on the next page
-        if ($style->counter_reset && ($reset = $style->counter_reset) !== "none") {
-            $vars = preg_split('/\s+/', trim($reset), 2);
-            $split->_counters['__' . $vars[0]] = $this->lookup_counter_frame($vars[0])->_counters[$vars[0]];
-        }
+        // Preserve the current counter values. This must be done after the
+        // parent split, as counters get reset on frame reset
+        $split->_counters = $this->_counters;
```

Comment:
This also handles cases where the split element does not reset the counter itself. The counter reset is then prevented on the split node by setting `counter-reset` to `none` above.